diode-server/reconciler/server.go

```diff
@@ -95,15 +94,15 @@ func (s *Server) RetrieveIngestionDataSources(_ context.Context, in *reconcilerp
 	filterByName := in.Name != ""
 
 	if filterByName {
-		if _, ok := s.apiKeys[in.Name]; !ok || !strings.HasPrefix(in.Name, "INGESTION") {
+		if _, ok := s.apiKeys[in.Name]; !ok || in.Name != "DIODE_API_KEY" {
```

Comment:
s/DIODE_API_KEY/DIODE, see api key names in `diode-server/reconciler/api_keys.go`